testing/upload_validation.ts

```diff
@@ -95,10 +96,26 @@ export class PackMetadataValidationError extends Error {
   }
 }
 
-export async function validatePackVersionMetadata(metadata: Record<string, any>): Promise<PackVersionMetadata> {
+export async function validatePackVersionMetadata(
+  metadata: Record<string, any>,
+  sdkVersion: string | undefined,
+): Promise<PackVersionMetadata> {
+  let combinedSchema: z.ZodType<Partial<PackVersionMetadata>> = legacyPackMetadataSchema;
+
+  // Server-side validation may be running a different SDK version than the pack maker
+  // is using, so some breaking changes to metadata validation can be set up to only
+  // take effect before or after an SDK version bump.
+  if (sdkVersion) {
```

Comment:
I guess sdkVersion is only absent for some very old pack versions?

Reply:
Looks like sdkVersion started being included in https://github.com/coda/packs-sdk/pull/970 in May 2021

Reply:
yeah, feel like we can just assume sdkVersion is always available.

Reply:
or actually, I am not sure if we have any packs that old. maybe we do. the current way is good. is it possible to add some log here?

Reply:
Not sure that here in the SDK wouldn't be the best place to log, but it'd be simple enough to logging or metrics directly from server code before calling into `validatePackVersionMetadata`.

(My preferred approach to tracking the use of deprecated features would probably be a workflow that scans all packs daily and gathers any deprecation warnings or other findings that might impact changes we're hoping to make)

Reply:
> Not sure that here in the SDK wouldn't be the best place to log, but it'd be simple enough to logging or metrics directly from server code before calling into validatePackVersionMetadata.

makes sense. 

> (My preferred approach to tracking the use of deprecated features would probably be a workflow that scans all packs daily and gathers any deprecation warnings or other findings that might impact changes we're hoping to make)

I still feel like it's easier to just query athena if we make all external metadata queryable. but a workflow idea definitely works.